code/game/objects/items/RSF.dm

```
@@ -8,178 +8,201 @@ RSF
+								/obj/item/paper = 10,
+								/obj/item/storage/pill_bottle/dice = 200,
+								/obj/item/pen = 50,
+								/obj/item/clothing/mask/cigarette = 10,
+								)
+	///An associated list of fuel and it's value
+	var/list/matter_by_item = list(/obj/item/rcd_ammo = 10,)
+	///A list of surfaces that we are allowed to place things on.
+	var/list/allowed_surfaces = list(/turf/open/floor, /obj/structure/table)
+	///The unit of mesure of the matter, for use in text
+	var/discriptor = "fabrication-units"
+	///The verb that describes what we're doing, for use in text
+	var/action_type = "Dispensing"
+
+/obj/item/rsf/Initialize()
+	. = ..()
+	to_dispense = cost_by_item[1]
 
 /obj/item/rsf/examine(mob/user)
 	. = ..()
-	. += "<span class='notice'>It currently holds [matter]/30 fabrication-units.</span>"
+	. += "<span class='notice'>It currently holds [matter]/[max_matter] [discriptor].</span>"
 
 /obj/item/rsf/cyborg
 	matter = 30
 
 /obj/item/rsf/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/rcd_ammo))
-		if((matter + 10) > 30)
-			to_chat(user, "<span class='warning'>The RSF can't hold any more matter!</span>")
+	if(is_type_in_list(W,matter_by_item))//If the thing we got hit by is in our matter list
+		var/tempMatter = matter_by_item[W.type] + matter
+		if(tempMatter > max_matter)
+			to_chat(user, "<span class='warning'>\The [src] can't hold any more [discriptor]!</span>")
 			return
 		qdel(W)
-		matter += 10
+		matter = tempMatter //We add its value
 		playsound(src.loc, 'sound/machines/click.ogg', 10, TRUE)
-		to_chat(user, "<span class='notice'>The RSF now holds [matter]/30 fabrication-units.</span>")
-		icon_state = "rsf"
+		to_chat(user, "<span class='notice'>\The [src] now holds [matter]/[max_matter] [discriptor].</span>")
+		icon_state = initial(icon_state)//and set the icon state to the initial value it had
 	else
 		return ..()
 
 /obj/item/rsf/attack_self(mob/user)
+	if(!user)
+		return
```

When would this proc be called without a `user`?
Add a `CRASH` instead of a `return` if you think this could ever happen, else you are making it silently error.

---

I'll get rid of it. I think it's either a legacy of the rsf or of the rcd. Not sure what it was put in to prevent.